### PR TITLE
Bipods

### PR DESCRIPTION
**Pull request by Caulaflower**
## Additions
- Bipod mechanics

## Changes

Adds a gizmo for bipod setup, which changes stats specificied in bipodcategorydef
Bipodcategorydefs are used in bipodinjector which handles their compatiblity, all guns need to get bipods is a weapon tag matching a bipodcategorydef ID

## Reasoning

- LMG need some nerfs, which shouldn't be just simple stat stuff
- DMRs and sniper rifles need the opposite

## Alternatives

Stat nerfing

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] Playtested a colony (specify how long)

### CODE REVIEW

**Review by albertbokor — COMMENTED** [review]
Niiice! Works fine at me!

File: Source/CombatExtended/CombatExtended/Settings.cs [hunk 1]
```diff
@@ -14,6 +14,8 @@ public class Settings : ModSettings
         #region Settings
 
         // General settings
+        private bool bipodMechanics = false;
```
Comment (N7Huntsman) [inline Source/CombatExtended/CombatExtended/Settings.cs:17]: Any reason not to have this turned on by default?

**Review by N7Huntsman — APPROVED** [review]
Appears to work as intended.

However, is there a way to add a 'progress bar' when setting up a bipod in a similar way to when reloading? 

Additionally, I think the current gizmo icon/label has the potential to be confusing. Rather than 'set up bipod' or 'close bipod' (with the inference that it's presently in the opposite state), perhaps it'd be better to change the label to 'Deploy Bipod' and make it like the 'Hold Fire' gizmo w/ a check mark or x-mark?

**Review by N7Huntsman — COMMENTED** [review]
Just for the sake of argument, is there any advantage to configuring the bipods this way (that is, with their own custom def) versus having all the same stuff in a comp or mod extension?

**Comment by N7Huntsman** [comment]
Additionally, adding something to the weapon stats that denotes that the weapon has a bipod may be useful to the player.

**Comment by Caulaflower** [comment]
> Just for the sake of argument, is there any advantage to configuring the bipods this way (that is, with their own custom def) versus having all the same stuff in a comp or mod extension?

I'm not an xml coder but my idea was that it's easier to patch this way, you can still technically patch guns one by one but I don't see much reason for it

**Comment by Caulaflower** [comment]
> Additionally, adding something to the weapon stats that denotes that the weapon has a bipod may be useful to the player. Perhaps it could include a general summary of the bipod stats, as well.

would that be enough? 
image

**Comment by ItsSamGray** [comment]
> > Additionally, adding something to the weapon stats that denotes that the weapon has a bipod may be useful to the player. Perhaps it could include a general summary of the bipod stats, as well.
> 
> would that be enough? image

Looks great! the red text needs fixing tho I think.

**Comment by ItsSamGray** [comment]
> > 
> 
> it's colored on purpose, but it can be easily changed

It being red is great, I meant the text that needs fixing. green is "when set up" and red must read "when not set up", right?

**Comment by Caulaflower** [comment]
> > > 
> > 
> > 
> > it's colored on purpose, but it can be easily changed
> 
> It being red is great, I meant the text that needs fixing. green is "when set up" and red must read "when not set up", right?

ah yeah I haven't noticed, gonna fix it and upload the changes in a few hours

**Review by albertbokor — APPROVED** [review]

**Comment by Caulaflower** [comment]
forgot to mention, but also added text about in which mods bipods autosetup with autosetup option on and fixed an issue where in manual setup mode "close bipod" option would leave it on

**Review by ItsSamGray — APPROVED** [review]
Looking great and working as intended on paper and ingame

**Review by perkinslr — DISMISSED** [review]
I think this is fine.  

As a note for the XML-side, generally a weapon should probably *only* get bonuses for a bipod, or *only* penalties for no bipod.  If a *particular* weapon gets warmup bonuses for bipod and warmup penalties for no bipod, then the weapon *never* gets its stats listed, and it could easily have the base value changed to either *final* value, and then only apply the opposite side.

(e.g. if the weapon has a base value of 2, and a 1 point bonus and 1 point penalty, just give it a base value of 1 and a 2 point bonus and no penalty, or a base value of 3 and a 2 point penalty *only*).

**Review by ItsSamGray — APPROVED** [review]
We're all set

**Pull request merged**